### ISSUE TITLE
Rename `annotations` to `decorators`

### ISSUE THREAD

**Issue by LestaD** [google/traceur-compiler#2078]
https://github.com/wycats/javascript-decorators
https://medium.com/google-developers/exploring-es7-decorators-76ecb65fb841#.tw5s3qr0m
http://blog.developsuperpowers.com/eli5-ecmascript-7-decorators/

**Comment by LestaD** [comment]
Because decorators `decorate` class and functions, not annotate:

```js
@Decorator
class Name {}
```

```js
var Name = (function(){
  var _Name = function(){};
  return Decorator(Name) || Name;
})()
```

**Comment by arv** [comment]
We also need to update these to match the latest proposal.
